added User entity, enum UserType, DTOs/user, UserRepository, UserService, UserServiceImpl, UserController, and DataInitializer

krub-dev opened the pull request:
This pull request implements the complete user management functionality for the IronBooking backend. The following components have been developed, and documented:

---

### User Entity and DTOs

- **User Entity:** Java model with fields `id`, `firstName`, `lastName`, `email`, `passwordHash`, and `userType` (enum: CLIENT/OWNER).
- **DTOs:** Request, response, and update DTOs with field validation using Jakarta Validation annotations.

---

### User Repository

- JPA repository for User with basic CRUD methods and custom queries (by email and by userType).

---

### Service Layer

- **UserService interface** and **UserServiceImpl** with all business logic:
  - Operations to create, read, update, and delete users.
  - Search by email and by user type.
  - Proper exceptions for not found and validation errors.

---

### User Controller

- **REST Controller** (`/users`) with endpoints:
  - `GET /users` - List all users
  - `GET /users/{id}` - Get user by id
  - `POST /users` - Create user (returns 201 + Location)
  - `PUT /users/{id}` - Update user
  - `DELETE /users/{id}` - Delete user
  - `GET /users/email/{email}` - Find user by email
  - `GET /users/type/{userType}` - List users by type

- **Validation:** Uses `@Valid`, `@Positive`, and appropriate HTTP status codes.

---

### Data Initialization

- **DataInitializer utility** (`CommandLineRunner`) to load sample users into the database if empty.
- Template sections prepared for Booking, Place, and Review entities to facilitate future team expansion.

---

### Documentation & Comments

- All classes and methods are clearly commented for maintainability and easy onboarding of new developers.
- Code follows project and Spring Boot best practices.
- The team can extend the same structure for Booking, Place, and Review entities.
- Future improvements: add integration tests, enhance validations, implement security.

> **Note:**  
> The current implementation stores user passwords as plain text for development and demonstration purposes only.  
> Once we cover Security in class, I will refactor this part to properly hash and securely store the `passwordHash` field according to best practices.

---

- [x] Entity, enum, DTOs, repository, service, and controller implemented
- [x] DataInitializer with sample data for User (for now)
- [x] Ready for review and merge

---

Please review and suggest any improvements!

AidaG91 commented on src/main/java/com/ironhack/IronBooking/controller/UserController.java:
```diff
@@ -0,0 +1,72 @@
+
+    private final UserService userService;
+
+    // Constructor injection for the service dependency
```
Replace by @Autowired

krub-dev commented:
## Changes

These are required changes requested during the review of an existing Pull Request.
- Modified the `User` entity to include a `@OneToMany` relationship with `Booking`.
- Added custom validation messages to fields in the `User`.
- Bookings management and display are only present in the `UserResponseDTO`.
- Added a method in `UserServiceImpl` to retrieve all bookings for a user, mapping each to `BookingResponseDTO` (implemented by Luis).
- Updated the `UserService` interface to include this method.
- Added the `GET /users/{id}/bookings` endpoint in `UserController` to expose a user's bookings.
- Added clarifying comments in the code.
- No breaking changes to existing endpoints.

AidaG91 reviewed (approved):
Todo ok :)
